xpreports/windows/managedinstalls.go

```diff
@@ -0,0 +1,150 @@
+package windows
+
+import (
+	"encoding/json"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// GetManagedInstalls is a crappy representation of things installed on a windows client
+func get32BitPrograms() ([]Program, error) {
+	cmd := exec.Command("powershell", "gp", "HKLM:\\Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\*", "|", "ConvertTo-Json")
+
+	// cmd.Stderr = os.Stderr
+	o, err := cmd.Output()
+	if err != nil {
+		return nil, errors.Wrap(err, "exec gp 32 bit programs")
+	}
+
+	var p []Program
+
+	if err := json.Unmarshal(o, &p); err != nil {
+		return nil, errors.Wrap(err, "failed unmarshalling 32 bit programs")
+	}
+
+	return p, nil
+}
+
+// GetManagedInstalls is a crappy representation of things installed on a windows client
+func get64BitPrograms() ([]Program, error) {
+	cmd := exec.Command("powershell", "gp", "HKLM:\\Software\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\*", "|", "ConvertTo-Json")
+
+	// cmd.Stderr = os.Stderr
+	o, err := cmd.Output()
+	if err != nil {
+		return nil, errors.Wrap(err, "exec gp 64 bit programs")
+	}
+
+	var p []Program
+
+	if err := json.Unmarshal(o, &p); err != nil {
+		return nil, errors.Wrap(err, "failed unmarshalling 64 bit programs")
+	}
+
+	return p, nil
+}
+
+// marshalManagedInstallFields takes a golang struct and uses field tags to conver to what sal wants
+func formatManagedInstallFields(p []Program) ([]byte, error) {
+
+	var all []ManagedInstallsFormatted
+
+	for _, element := range p {
+		all = append(all, ManagedInstallsFormatted{
+			Description:      element.Description,
+			DisplayName:      element.DisplayName,
+			Installed:        element.Installed,
+			InstalledSize:    element.InstalledSize,
+			InstalledVersion: element.InstalledVersion,
+			Name:             element.Name,
+		})
+	}
+
+	data, err := json.MarshalIndent(&all, "", "  ")
+	if err != nil {
+		return nil, errors.Wrap(err, "message")
+	}
+
+	return data, nil
+}
+
+// filterManagedInstalls strips out things "apps and programs" wouldn't report to the user (i.e MS Patches, etc)
+func filterManagedInstalls(p []Program) ([]Program, error) {
+
+	var filtered []Program
+
+	for _, element := range p {
+		if element.DisplayName == "" || strings.Contains(element.Name, "0FF1CE") {
+			continue
+		} else {
```

Comment:
when `continue` is called there rest of the function is not executed. Close the `if {}` and drop the else to remove extra indendation.